src/atoms/Switch/Switch.native.js

```diff
@@ -0,0 +1,315 @@
+
+const styles = {
+  size({ size, theme }) {
+    switch (size) {
+      case 'medium':
+      default:
+        return {
+          containerSize: {
+            height: theme.spacings.large,
+            width: theme.spacings.xxxlarge,
+          },
+          knobSize: {
+            height: theme.spacings.medium,
+            width: theme.spacings.medium,
+            activeWidth: theme.spacings.large,
+          },
+        };
+      case 'large':
+        return {
+          containerSize: {
+            height: makePxValue(3),
+            width: makePxValue(6),
+          },
+          knobSize: {
+            height: theme.spacings.large,
+            width: theme.spacings.large,
+            activeWidth: theme.spacings.xxlarge,
+          },
+        };
+    }
+  },
+  padding({ size, theme }) {
+    switch (size) {
+      case 'medium':
+      default:
+        return theme.spacings.xxsmall;
+      case 'large':
+        return theme.spacings.xsmall;
+    }
+  },
+  alignSelf({ align }) {
+    switch (align) {
+      case 'left':
+      default:
+        return 'flex-start';
+      case 'center':
+        return 'center';
+      case 'right':
+        return 'flex-end';
+    }
+  },
+};
+
+const StyledContainer = styled(TouchableOpacity)(
+  (props) => `
+  height: ${styles.size(props).containerSize.height};
+  width: ${styles.size(props).containerSize.width};
+  border-radius: ${`${parseInt(styles.size(props).containerSize.height, 10) / 2}px`};
+  padding: ${styles.padding(props)};
+  flex-direction: row;
+  align-self: ${styles.alignSelf(props)};
+`,
+);
+
+const StyledKnob = styled(View)(
+  (props) => `
+  position: absolute;
+  align-self: center;
+  height: ${styles.size(props).knobSize.height};
+  border-radius: ${`${parseInt(styles.size(props).knobSize.height, 10) / 2}px`};
+  background-color: #fff;
+`,
+);
+
+const AnimatedContainer = Animated.createAnimatedComponent(StyledContainer);
+const AnimatedKnob = Animated.createAnimatedComponent(StyledKnob);
+
+const moveKnob = (config, cb) => {
+  const [leftSpaceAnim, rightSpaceAnim] = config;
+  Animated.parallel([
+    Animated.timing(leftSpaceAnim.animationProp, {
+      toValue: leftSpaceAnim.toValue,
+      duration: leftSpaceAnim.duration,
+    }),
+    Animated.timing(rightSpaceAnim.animationProp, {
+      toValue: rightSpaceAnim.toValue,
+      duration: rightSpaceAnim.duration,
+    }),
+  ]).start(cb);
+};
+
+const Switch = (props) => {
+  const { disabled, on, defaultOn, onChange, size, align, testID } = props;
+  const theme = useContext(ThemeContext);
+  const {
+    containerSize: { width: containerWidth },
+    knobSize: { width: knobWidth, activeWidth: knobActiveWidth },
+  } = styles.size({ size, theme });
+  const containerPadding = parseInt(styles.padding({ size, theme }), 10);
+  const [toggleState, setToggleState] = useState(false);
+  const [active, setActive] = useState(false);
+
+  const SWITCH_STATE_CONFIGS = {
+    OFF_LEFT_SPACE: containerPadding,
```

Comment:
If any AnimationConfig is not straight-forward value like ```0``` or ```100```, can we have a getter function that resolves it? This makes the ```AnimationConfig``` look cleaner and refactoring the animation becomes easier